docs/reference/plugins/colcon_plugin.rst

```diff
@@ -0,0 +1,127 @@
+.. _reference_colcon_plugin:
+
+Colcon plugin
+=============
+
+The ``colcon`` plugin is useful when building `ROS 2 <http://www.ros.org/>`_ parts
+that use the `colcon <https://colcon.readthedocs.io/en/released/>`_ build tool.
+
+Keywords
+--------
+
+In addition to the common :ref:`plugin <part-properties-plugin>` and
+:ref:`sources <part-properties-sources>` keywords, this plugin provides the following
+plugin-specific keywords:
+
+colcon-ament-cmake-args
+~~~~~~~~~~~~~~~~~~~~~~~
+**Type:** list of strings
+**Default:** []
+
+Arguments to pass to ament_cmake packages. Note that any arguments here that match
+colcon arguments need to be prefixed with a space. This can be done by quoting each
+argument with a leading space.
+
+colcon-catkin-cmake-args
+~~~~~~~~~~~~~~~~~~~~~~~~
+**Type:** list of strings
+**Default:** []
+
+Arguments to pass to catkin packages. Note that any arguments here which match colcon
+arguments need to be prefixed with a space. This can be done by quoting each argument
+with a leading space.
+
+colcon-cmake-args
+~~~~~~~~~~~~~~~~~
+**Type:** list of strings
+**Default:** []
+
+Arguments to pass to cmake projects. Note that any arguments here which match colcon
+arguments need to be prefixed with a space. This can be done by quoting each argument
+with a leading space.
+
+colcon-packages
+~~~~~~~~~~~~~~~
+**Type:** list of strings
+**Default:** []
+
+List of colcon packages to build. If not specified, all packages in the workspace will
+be built. If set to an empty list (``[]``), no packages will be built, which could
+be useful if you only want ROS Debian packages in the snap.
```

Comment:
```suggestion
be useful if you only want Debian packages in the snap.
```

And now we run into an issue with this phrase. "ROS Debian" has the potential to sound like a distribution.

Reply:
https://github.com/canonical/snapcraft/pull/5470/commits/2189577036587bfc12298975212c4b2fb473a4b3